packages/uhk-web/src/app/store/effects/contributors.effect.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Actions, Effect } from '@ngrx/effects';
+import { Action, Store } from '@ngrx/store';
+
+import { Observable } from 'rxjs/Observable';
+import { from } from 'rxjs/observable/from';
+import { of } from 'rxjs/observable/of';
+import { map, switchMap, catchError, reduce, mergeMap, take, withLatestFrom } from 'rxjs/operators';
+
+import { Constants } from 'uhk-common';
+
+import { State } from './../reducers/contributors.reducer';
+import { AppState, contributors } from '../index';
+import { UHKContributor } from '../../models/uhk-contributor';
+import {
+    AgentContributorsAvailableAction,
+    AgentContributorsNotAvailableAction,
+    GetAgentContributorsAction,
+    ActionTypes,
+    FetchAgentContributorsAction
+} from '../actions/contributors.action';
+
+@Injectable()
+export class ContributorsEffect {
+    @Effect() getContributors$: Observable<Action> = this.actions$
+        .ofType<GetAgentContributorsAction>(ActionTypes.GET_AGENT_CONTRIBUTORS)
+        .pipe(
+            withLatestFrom(this.store.select(contributors)),
+            map(([action, state]) => {
+                if (state.contributors.length === 0) {
+                    return new FetchAgentContributorsAction();
+                }
+                return new AgentContributorsAvailableAction(state.contributors);
+            })
+        );
+
+    @Effect() fetchContributors$: Observable<Action> = this.actions$
+        .ofType<FetchAgentContributorsAction>(ActionTypes.FETCH_AGENT_CONTRIBUTORS)
+        .pipe(
+            mergeMap(() => this.http.get<UHKContributor[]>(Constants.AGENT_CONTRIBUTORS_GITHUB_API_URL)),
+            switchMap((response: UHKContributor[]) => {
+                return from(response).pipe(
+                    mergeMap(
+                        (contributor: UHKContributor) => {
+                            return this.http.get(contributor.avatar_url, { responseType: 'blob' });
```

Comment:
Why do you load the avatar images? The electron renderer process can load the images.

Reply:
Because it renders awfully. Images show up randomly for contributors, so I decided to wait until each is available before applying the template.

Reply:
Ok. I checked, ~0.5kb the size of an image, it is not big